lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1229,7 +1275,37 @@ func (d *zfs) RefreshVolume(vol Volume, srcVol Volume, srcSnapshots []Volume, al
 
 // DeleteVolume deletes a volume of the storage device. If any snapshots of the volume remain then
 // this function will return an error.
+// For image volumes, both filesystem and block volumes will be removed.
 func (d *zfs) DeleteVolume(vol Volume, op *operations.Operation) error {
+	for _, suffix := range blockBackedAllowedFilesystems {
+		tmpVol := vol
```

Comment:
Can you add `Clone()` function to `Volume` type and then use the same logic that `lxdBackend.GetVolume()` uses (i.e pre-allocating the map size, as its more efficient), and then update `lxdBackend.GetVolume()` to just call:

```go
return drivers.NewVolume(b.driver, b.name, volType, contentType, volName, newConfig, newPoolConfig).Clone()
```

Then you can also use `tmpVol := vol.Clone()` here.

Additionally I don't think we need to clone the `tmpVol` on every iteration, but instead just do it once before the loop starts and then modify the same tmpVol for each filesystem iteration.